007-OFEP-transaction-context-propagation.md

```diff
@@ -0,0 +1,91 @@
+## Transaction Context Propagation
+
+## State: DRAFTING
+
+Evaluation context is a container for all contextual data that can be used used to conditionally control the value returned from a flag evaluation. Traditionally, a developer would be responsible for explicitly defining evaluation context during flag evaluation.
+
+Transaction context, a subset of the evaluation context, is a container for transaction specific data (e.g. user id, user agent, IP) that lives for the duration of a transaction. With transaction context propagation, a developer can set transaction context where it's convenient (e.g. an auth service) and have it persist for the length of a transaction (e.g. a request or thread). OpenFeature will then merge transaction context with evaluation context before flag evaluation.
```

Comment:
```suggestion
Flag evaluation may be affected by [evaluation context](link here) data including global, client, invocation, and hook context.
Traditionally, a developer would be responsible for explicitly defining evaluation context during flag evaluation.
This proposal defines a new type of evaluation context called transaction context.

Transaction context is a container for transaction-specific evaluation context (e.g. user id, user agent, IP).
With transaction context propagation, a developer can set transaction context where it's convenient (e.g. an auth service) and it will automatically be applied to all flag evaluations within a transaction.
Transaction context will typically be propagated using a language-specific carrier such as thread local storage or another similar mechanism.
```